src/target/source/codegen_source_base.h

```diff
@@ -155,7 +156,8 @@ runtime::Module CSourceModuleCreate(const String& code, const String& fmt,
  */
 runtime::Module CreateMetadataModule(
     const std::unordered_map<std::string, runtime::NDArray>& params, runtime::Module target_module,
-    const Array<runtime::Module>& ext_modules, Target target);
+    const Array<runtime::Module>& ext_modules, Target target,
```

Comment:
given this is just invoked from build_module.cc, maybe we should make metadata have no default?

Reply:
Actually, doesn't seem a bad idea to remove the default and add an empty metadata in `compiler.cc`, so it is clear that we don't need compiled metadata in the vm executor

Reply:
great, i agree with that logic